app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -360,7 +369,15 @@ private fun ConversationScreen(
             SelfDeletionMenuItems(
                 hideEditMessageMenu = conversationScreenState::hideContextMenu,
                 currentlySelected = messageComposerState.getSelfDeletionTime(),
-                onSelfDeletionDurationChanged = { messageComposerState.specifySelfDeletionTime(it) }
+                onSelfDeletionDurationChanged = {
+                    val newSelfDeletingMessagesStatus = SelfDeletingMessagesStatus(
+                        isFeatureEnabled = true, // Otherwise the menu wouldn't be visible
+                        hasFeatureChanged = false,
+                        globalSelfDeletionDuration = it.toSeconds(),
+                        isEnforced = false
+                    )
+                    onNewSelfDeletingMessagesStatus(newSelfDeletingMessagesStatus)
```

Comment:
since the only changing variable here is globalSelfDeletionDuration perhaps a better idea would be to pass only this variable to onNewSelfDeletingMessagesStatus and create SelfDeletingMessagesStatus once we propage it to kalium ?

Reply:
Since this is a grouping class it doesn't really matter whether we pass one object or the other, but sure, I can change it like that.